php/Plugin.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Plugin Utility class.
+ *
+ * @package phpdocplugin
+ */
+
+namespace Pods\phpdocplugin;
+
+/**
+ * Plugin utility class.
+ */
+class Plugin {
+
+	/**
+	 * Absolute path to the main plugin file.
+	 *
+	 * @var string
+	 */
+	protected $file;
+
+	/**
+	 * Absolute path to the root directory of this plugin.
+	 *
+	 * @var string
+	 */
+	protected $dir;
+
+	/**
+	 * Setup the plugin.
+	 */
+	public function __construct() {
+		$this->file = PHPDOCPLUGIN_FILE;
+		$this->dir  = dirname( PHPDOCPLUGIN_FILE );
+	}
+
+	/**
+	 * Plugin init function.
+	 *
+	 * @see WP init action.
+	 */
+	public function init() {
+		// todo - init actions here.
+	}
+
+	/**
+	 * Is WP debug mode enabled.
+	 *
+	 * @return boolean
+	 */
+	public static function is_debug() {
+		return ( defined( 'WP_DEBUG' ) && WP_DEBUG );
+	}
+
+	/**
+	 * Is WP script debug mode enabled.
+	 *
+	 * @return boolean
+	 */
+	public static function is_script_debug() {
+		return ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG );
+	}
+
+	/**
+	 * Return the current version of the plugin.
+	 *
+	 * @return mixed
+	 */
+	public function version() {
+		return $this->meta( 'Version' );
+	}
+
+	/**
+	 * Sync the plugin version with the asset version.
+	 *
+	 * @return string
+	 */
+	public static function asset_version() {
+		if ( self::is_debug() || self::is_script_debug() ) {
+			return time();
+		}
+
+		return self::get_plugin_version();
+	}
+
+	/**
+	 * Get plugin meta data.
+	 *
+	 * @param  string $field Optional field key.
+	 *
+	 * @return array|string|null
+	 */
+	public function meta( $field = null ) {
+		static $meta;
+
+		if ( ! isset( $meta ) ) {
+			$meta = get_file_data( $this->file, array(), $field );
+		}
+
+		if ( isset( $field ) ) {
+			if ( isset( $meta[ $field ] ) ) {
+				return $meta[ $field ];
+			}
+
+			return null;
+		}
+
+		return $meta;
+	}
+
+	/**
+	 * Return the plugin slug.
+	 *
+	 * @return string plugin slug.
+	 */
+	public static function get_plugin_slug() {
+		return dirname( plugin_basename( PHPDOCPLUGIN_FILE ) );
+	}
+
+	/**
+	 * Return the basefile for the plugin.
+	 *
+	 * @return string base file for the plugin.
+	 */
+	public static function get_plugin_file() {
+		return plugin_basename( PHPDOCPLUGIN_FILE );
+	}
+
+	/**
+	 * Return the version for the plugin.
+	 *
+	 * @return float version for the plugin.
+	 */
+	public static function get_plugin_version() {
+		return PHPDOCPLUGIN_VERSION;
+	}
+
+	/**
+	 * Returns appropriate html for KSES.
+	 *
+	 * @param bool $svg Whether to add SVG data to KSES.
+	 */
+	public static function get_kses_allowed_html( $svg = true ) {
```

Comment:
Is this necessary?

Reply:
No, just filler from a plugin I wrote. I'll remove it.